python/lsst/sims/maf/stackers/ditherStackers.py

```diff
@@ -29,92 +42,181 @@ def wrapRA(ra):
     ra = ra % (2.0*np.pi)
     return ra
 
-class RandomDitherStacker(BaseStacker):
+def inHexagon(xOff, yOff, maxDither):
+    """
+    Identify dither offsets which fall within the inscribed hexagon.
+    """
+    # Set up the hexagon limits.
+    #  y=mx+b, 2h is the height.
+    m = np.sqrt(3.0)
+    b = m*maxDither
+    h = m/2.0*maxDither
+    # Identify offsets inside hexagon.
+    inside = np.where((yOff < m*xOff + b) &
+                      (yOff > m*xOff - b) &
+                      (yOff < -m*xOff + b) &
+                      (yOff > -m*xOff -b ) &
+                      (yOff < h) & (yOff > -h))[0]
+    return inside
+
+
+
+def polygonCoords(nside, radius, rotationAngle):
     """
-    Randomly dither the RA and Dec pointings up to maxDither degrees from center, different offset per visit.
+    Find the x,y coords of a polygon. (useful for plotting dither points).
     """
-    def __init__(self, raCol='fieldRA', decCol='fieldDec', maxDither=1.75, randomSeed=None):
+    eachAngle= 2*np.pi/nside
+    xCoords= np.zeros(nside, float)
+    yCoords= np.zeros(nside, float)
+    for i in range(0,nside):
+        xCoords[i]= np.sin(eachAngle*i + rotationAngle)*radius
+        yCoords[i]= np.cos(eachAngle*i + rotationAngle)*radius
+    return zip(xCoords,yCoords)
+
+
+class RandomDitherFieldVisitStacker(BaseStacker):
+    """
+    Randomly dither the RA and Dec pointings up to maxDither degrees from center, different offset for each field, for each visit.
+    Optionally constraint dither offsets to lie within inscribed hexagon of FOV.
+    """
+    def __init__(self, raCol='fieldRA', decCol='fieldDec', maxDither=1.75,
+                 inHex=True, randomSeed=None):
```

Comment:
Given how much confusion there always is about where to use radians and where to use degrees, should there be a docstring here explaining that maxDither is in degrees?  (I guess that would go for all the other new classes, as well)